docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+
+### CSS styles
+
+- Option A - Continue using ad-hoc inline styles
+- Option B - Use TailwindCSS
+- Option C - Use CSS-in-JS
+- Option D - Use a CSS pre-processor like SASS
+
+### Email previewing
+
+- Option A - Continue to use the `write-emails` command
+- Option B - Use Storybook
+- Option C - Use Mailtrap
+
+--
+
+Furthermore, there are a few **other decisions** worth noting that won’t necessarily have pros/cons lists:
+
+- How to handle generating plaintext versions
+- Transitioning to Fluent over GetText for localization
+- Plans around integration involving feature flagging and the QA process
+
+## Decision Outcomes
+
+### Templating
+
+Chosen option: "Option B - Use React server-side to generate static HTML templates", because:
+
+- React is being used in other parts of the FxA ecosystem and using it as part of `fxa-auth-server` frontend stack will lead to consistency across FxA
+- It has been tried and tested to work well with Storybook
+- Components across the templates can be reused
+- We'd be able to accomodate logic easily
+
+A note about tests: Since we're going with the React-based solution for our templating stack, it may make sense to use Jest to test the templates. However, we don't want a ton of time spent in test land. At the very least, we can take what we currently have and transfer/copy it, and look them over to see if anything can be improved without a high level of effort.
+
+#### MJML
+
+We have decided to not use MJML as part our templating stack, because:
+
+- MJML's claim is to code responsive emails easily but FxA emails are currently not designed to need much in terms of responsiveness and it'd add a layer of complication we wouldn't benefit much from during a mustache to React conversion.
+- If we would like to redesign our templates and make them responsive at a later point, we can re-evaluate this choice.
+
+### CSS Styles
+
+Chosen option: Use TailwindCSS, because:
+
+- It is already being used in `fxa-settings` so the styles can be shared across the services
+- Tailwind classes can be converted into inline styles to ensure maximum compatibility of our emails.
+- It is highly configurable leading to the possibility of reusing the config file setup in `fxa-settings` for our templates
+- With tailwind's utility and compononent classes, we can DRY up our codebase
+
+### Email Previewing
+
+Chosen option: "Option B - Use Storybook", because:
+
+- It is easy to setup and provides us the flexibility to preview all of the different email states together.
+- We have a Storybook deployment already in place and hooking it up for `fxa-auth-server` will ensure consistency.
+- Easy to test and perform QA without needing to touch the codebase.
+- Support for TailwindCSS
+
+Note: Along with using storybook to view the various states of the templates, we are also planning on continued support of [Maildev](https://www.npmjs.com/package/maildev) since it has applications beyond just previewing emails.
+
+### Other
```

Comment:
This set of choices, though being the most consistent with the dependency choices we use elsewhere seems to me to have the most custom and complex build. It's also a pretty drastic change. The consistency of React/Tailwind/Storybook is appealing but I still wonder if that benefit carries enough weight compared to the other options.

Reply:
Yeah, I do think we should note that in the cons of using this stack. I don't think this ADR has a positive/negative consequences section but if we did end up going with this, we should add one even if just to note that this is one of the more complex options and will require some additional build steps.